mirrord/layer/src/socket/ops.rs

```diff
@@ -569,7 +639,17 @@ pub(super) fn connect(
 
     let unix_streams = crate::setup().remote_unix_streams();
 
-    trace!("in connect {:#?}", SOCKETS);
+    let socket_fd_flags = nix::fcntl::fcntl(sockfd, nix::fcntl::FcntlArg::F_GETFL)
```

Comment:
Why do we need to check it?
If socket is blocking, it will block the app, if not it won't (and we'll get einprogress) - why does intproxy care?

Reply:
Because we block to get response from intproxy before we call `connect` to intproxy interceptor addr so my solution was to send einprogress as soon as intproxy handles the connect request and not when the reply from the agent is returned

Reply:
but you can just call the original `CONNECT` and if you get einprogress then you know it's not blocking and you pass it to caller and if it doesn't return it then you need to wait for a reply from upstream

Reply:
But connect to what address? intproxy doesn't yet start the interceptor until it has a response from the agent

Reply:
change order - start interceptor before getting answer from agent.

Reply:
(but don't accept until you get response, to simulate real stuff)

Reply:
What to do with errors? like when error is an actual error?

Reply:
Like right now I also don't have a good solution

Reply:
what happens when einprogress errors? I think it doesn't?
the internal proxy should just timeout it's socket/accept

Reply:
I also don't think we need this?